### PR TITLE
Basics of MySQL

### PR DESCRIPTION
**Pull request by happycoder0011**

### CODE REVIEW

**Comment by happycoder0011** [comment]
Kindly review.

**Comment by sudhanshutiwari264** [comment]
@happycoder0011
Sorry for the delay, kindly edit your PR description and follow [this](https://github.com/Crio-Bytes/Miscellaneous/blob/main/.github/PULL_REQUEST_TEMPLATE/adding-microbyte.md) template for the same in case of new MicroBytes. 
Also, your PR is under review.

**Review by archithdwij — CHANGES_REQUESTED** [review]
@happycoder0011 Your MicroByte is beginner friendly. But, try adding more activities to add scope for learners. The more doing there is, the more they'll learn. 
All the best 👍

**Review by Kevinpaulose05 — CHANGES_REQUESTED** [review]
Firstly make a proper PR description as mentioned by @sudhanshutiwari264 before.
The topic you have chosen is a fundamental one. Your purpose of making the microbyte is no to make ppl understand the basics of SQL by following some steps u give. Instead, they shud engage themselves by searching the web and following ur material o learn it by a hands-on approach, which we call as Learn by doing approach. Ur microbyte should have activities that have some basic guidelines/hints that ppl will follow and work their way thru to complete the activity and in the process learn. 
Follow the structure of readme(as mentioned in the google sheet(, introduction, background,pre-requisites and then activities. Refer to other microbytes to get the idea.
Also, ur microbyte is only text as of now. Add images for visual appeal and ease. 
U have given code for SELECT command only. All other like WHERE,HAVING,etc. are just mentioned. Give resource links and elaborate about them too. 
Kindly refrain from using -> as a bullet. Use proper markdown and structure ur content well; don't compress it into one single line.
U need to make ppl do/build something at the end of each activity. U may choose some dataset and show expected query output and give hints on what the code should contain to get that output. Pls add such activities.

**Comment by Kevinpaulose05** [comment]
@happycoder0011 Pls work on my comments. We are eager to see ur completed contribution. U will get  creator badges only if u do the needed latest by today midnight

**Comment by happycoder0011** [comment]
> @happycoder0011 Pls work on my comments. We are eager to see ur completed contribution. U will get creator badges only if u do the needed latest by today midnight

I have made the changes. Kindly, review.

**Comment by YourAKShaw** [comment]
> @happycoder0011
> Sorry for the delay, kindly edit your PR description and follow [this](https://github.com/Crio-Bytes/Miscellaneous/blob/main/.github/PULL_REQUEST_TEMPLATE/adding-microbyte.md) template for the same in case of new MicroBytes.
> Also, your PR is under review.

Kindly do this ASAP!

File: MySQL_basic_commands/README.md
Comment (amoghrajesh) [inline MySQL_basic_commands/README.md]: not* - typo

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @happycoder0011!
It is good content but I have a few things to request from you.
These are the changes:

1. You have added a Prerequisite section and never ended it. So does point 2, which is 'Create Database' come under it? Kindly correct this.

2. I don't like the fact that you have provided exact commands for all your content. For improvement what you can do is - for eg: Instead of giving exact command for say 'Insert', you can say 'For inserting values, insert command can be used' and provide some reference link to it. Then add maybe a few questions pertaining to it to check the learners understanding. 
I am saying this because we want a 'Learn By Doing' method and not a tutorial. What you have created is a tutorial. Don't get me wrong but if you have this tutorial, why would someone come here as compared to the overrated 'Coursera' or 'Udacity' Courses on Mysql?
(What I said for insert commands applied to all the commands that you have)

3.  You have provided a section saying - "Learn about BLOB, SET and ENUM". How should they learn all this, why is it important, where is it used. All these questions should be handled by your MicroByte. Kindly add references to all these. 

4. You have provided more examples than needed for insert, select, and alter. Cut them down and probably add questions/references.

5. You are not supposed to make any changes to `CONTRIBUTING.md` even though it is something as minor as adding a line break. Kindly remove that from the PR.
 
I think you can make these changes quickly since you already have the content.
Make them asap and we shall review it again.

File: MySQL_basic_commands/README.md
Comment (happycoder0011) [reply MySQL_basic_commands/README.md]: I got it. But dont you think giving link for every individual command would be a lot scattered and disrupting the learning. That's why I gave all the content part in the prerequisites. The activities below can be done only if they learn through the content.
The extra bytes(blob, enum). I am adding more references to that. 
Kindly confirm should I remove all the command part. I can add  references to w3school if you say.
Thanks.

Comment (amoghrajesh) [reply MySQL_basic_commands/README.md]: Sure makes sense. Maybe you can cut down the number of examples by a bit

**Review by Kevinpaulose05 — CHANGES_REQUESTED** [review]
Make the following changes and u will be pretty much good to go then:

1. Limit ur pre-requisite function to SQL installation. Creating database and all can be done under say "activity 0: Getting started". Pre-requisite should only be something a newbie can see to understand if the project is for him/her. Also hide the terminal image as a hidden dropdown.
2. For each command u have given out the direct code like "for example" and so. Hide them as hints using `details` markdown
3. Scrap ur label 1,2,3 and instead make them as activity 1,2,3. And make ur activity as a challenge and its subsequent levels as tasks 1,2,3.
4. Add some articles about relational database(SQL), commands used and also give some real time uses of these.

Pls do 1,2,3 ASAP and u may take time for 4.

File: MySQL_basic_commands/README.md
Comment (Kevinpaulose05) [inline MySQL_basic_commands/README.md]: Formatting error

File: MySQL_basic_commands/README.md
Comment (Kevinpaulose05) [inline MySQL_basic_commands/README.md]: Use the '>' tag for notes. To gray shade them

**Review by Kevinpaulose05 — CHANGES_REQUESTED** [review]
Adding to these, a couple more:
1. In second terminal image of `describe student` command mention there is an alternative way as heading to diff the photos.
2. Use proper formatting of bullets. Like after a heading like `Insert values in table`  use bullets for the two subsequent cases. Do this for all such cases. Also, use a colon(:) after the heading separating it from its description.
3. Before ur tasks mention the heading as a challenge, then sub-headings as tasks.
5. Don't mention `extra bytes`, call it anything else like trivia/test urself/etc. What u r creating is a microbyte. Within that don't mention "bytes"
6. Center align all tables
7. Give a brief 2-3 word heading to each activity(tasks also if possible)

Most importantly change ur repo name, give a professional one like the one u gave for this PR

**Comment by Kevinpaulose05** [comment]
> Adding to these, a couple more:
> 
> 1. In second terminal image of `describe student` command mention there is an alternative way as heading to diff the photos.
> 2. Use proper formatting of bullets. Like after a heading like `Insert values in table`  use bullets for the two subsequent cases. Do this for all such cases. Also, use a colon(:) after the heading separating it from its description.
> 3. Before ur tasks mention the heading as a challenge, then sub-headings as tasks.
> 4. Don't mention `extra bytes`, call it anything else like trivia/test urself/etc. What u r creating is a microbyte. Within that don't mention "bytes"
> 5. Center align all tables

Kindly do these ASAP

File: CONTRIBUTING.md
Comment (YourAKShaw) [inline CONTRIBUTING.md:109]: Pls do not edit the `CONTRIBUTING.md` . We prohibit that. There's a minor change in it, but still. Kindly delete that from the PR

**Comment by Kevinpaulose05** [comment]
We have given you the label seeing the potential in ur PR and what we have achieved together. Kindly wrap-up ur microbyte with respect to our last comments. We are making an exception for you; u can complete this by maximum tomorrow night. After that we are sorry but we will have to close this PR which we don't want to since this is quality work. Kindly do the necessary ASAP and u can still get the creator badges and be considered for the other perks!

File: CONTRIBUTING.md
Comment (happycoder0011) [reply CONTRIBUTING.md:109]: How can i do that? Kindly help.

**Comment by happycoder0011** [comment]
> > Adding to these, a couple more:
> > 
> > 1. In second terminal image of `describe student` command mention there is an alternative way as heading to diff the photos.
> > 2. Use proper formatting of bullets. Like after a heading like `Insert values in table`  use bullets for the two subsequent cases. Do this for all such cases. Also, use a colon(:) after the heading separating it from its description.
> > 3. Before ur tasks mention the heading as a challenge, then sub-headings as tasks.
> > 4. Don't mention `extra bytes`, call it anything else like trivia/test urself/etc. What u r creating is a microbyte. Within that don't mention "bytes"
> > 5. Center align all tables
> 
> Kindly do these ASAP

I have made all the changes you noted. How to center the table?

File: Basics of MySQL/README.md
Comment (Kevinpaulose05) [inline Basics of MySQL/README.md]: Formatting error

File: Basics of MySQL/README.md
Comment (Kevinpaulose05) [inline Basics of MySQL/README.md]: mention microbyte in place of byte

**Review by Kevinpaulose05 — COMMENTED** [review]
Leave the table as it is. No issues. I assumed u used an img as the table. I don't think GitHub permits formatting of tables directly.

**Review by Kevinpaulose05 — APPROVED** [review]
Great work @happycoder0011 . Happy we could guide u. Another advocate will be shortly reviewing ur PR one last time and after their approval, we will merge it. Two things to do **after merging only:**
1. Open as many learners issues as u can(within the scope of ur microbyte)
2. Showcase ur newly merged microbyte in the #mirobyte-showcase channel in slack using this [template](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md)

Also, fill this [form](https://docs.google.com/forms/d/e/1FAIpQLSe8owYPVbIeaLJPV8vBHOsZxemRH8ZAk1srLnOkHzMt714yfA/viewform) right away.


Happy Contributing!!

**Comment by happycoder0011** [comment]
> Great work @happycoder0011 . Happy we could guide u. Another advocate will be shortly reviewing ur PR one last time and after their approval, we will merge it. Two things to do **after merging only:**
> 
> 1. Open as many learners issues as u can(within the scope of ur microbyte)
> 2. Showcase ur newly merged microbyte in the #mirobyte-showcase channel in slack using this [template](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md)
> 
> Also, fill this [form](https://docs.google.com/forms/d/e/1FAIpQLSe8owYPVbIeaLJPV8vBHOsZxemRH8ZAk1srLnOkHzMt714yfA/viewform) right away.
> 
> Happy Contributing!!

Thanks a lot for your efforts, patience and guidance throughout the process.

**Review by amoghrajesh — APPROVED** [review]
Hey @happycoder0011!
Thanks for working with us by making timely changes even after the challenge.
I appreciate all the effort you have put in by adding tables, images etc.

This MicroByte can be used a "getting started" guide for needy learners.
Also thanks for reverting the changes to `CONTRIBUTING.md`. It would make it hard to manage without that.

Cheers 😄

**Comment by happycoder0011** [comment]
> Hey @happycoder0011!
> Thanks for working with us by making timely changes even after the challenge.
> I appreciate all the effort you have put in by adding tables, images etc.
> 
> This MicroByte can be used a "getting started" guide for needy learners.
> Also thanks for reverting the changes to `CONTRIBUTING.md`. It would make it hard to manage without that.
> 
> Cheers 😄

Thanks a lot!!

**Comment by YourAKShaw** [comment]
@all-contributors kindly add @happycoder0011 for:

- [x] code
- [x] content
- [x] ideas
- [x] doc

**Pull request merged**
